### PR TITLE
ED-1225: Add oci oss support for Build/Core/Lms - Superseed PR 524

### PR DESCRIPTION
**Pull request by ddevadat**
updated cloud-store-sdk to 1.4.6 for supporting oci oss along with supporting class changes
Type of change

Please choose appropriate options.

    Bug fix (non-breaking change which fixes an issue)
    New feature (non-breaking change which adds functionality)
    Breaking change (fix or feature that would cause existing functionality to not work as expected)
    This change requires a documentation update

How Has This Been Tested?

Please describe the tests that you ran to verify your changes in the below checkboxes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

    Ran build job Build/Core/Lms

### CODE REVIEW

**Comment by maheshkumargangula** [comment]
@reshmi-nair - Please check this PR.

File: course-mw/sunbird-util/sunbird-platform-core/common-util/src/main/java/org/sunbird/common/models/util/oci/OciConnectionManager.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
+package org.sunbird.common.models.util.oci;
```
Comment (AmiableAnil) [inline course-mw/sunbird-util/sunbird-platform-core/common-util/src/main/java/org/sunbird/common/models/util/oci/OciConnectionManager.java:1]: Why this class is required here.

Comment (ddevadat) [reply course-mw/sunbird-util/sunbird-platform-core/common-util/src/main/java/org/sunbird/common/models/util/oci/OciConnectionManager.java]: originally this was added as there were csp specific implementation as part of tag release-5.0.0_RC2.  I think in the later revisions this was removed ( tag release-5.0.1_RC3)
i moved this oci folder from this pr

File: course-mw/sunbird-util/sunbird-platform-core/common-util/pom.xml [hunk 2]
```diff
@@ -267,9 +267,14 @@
...
     <repositories>
-        <repository>
+       <!-- <repository>
             <id>cloud-store</id>
             <url>https://oss.sonatype.org/content/repositories/orgsunbird-1021</url>
+        </repository> -->
+<!-- Below repository added for testing -->
+        <repository>
+            <id>cloud-store</id>
+            <url>https://oss.sonatype.org/content/repositories/orgsunbird-1144/</url>
```
[the remark this answers is not in the document]
Comment (ddevadat) [reply course-mw/sunbird-util/sunbird-platform-core/common-util/pom.xml]: removed it as it was a temporary measure, when cloud-store-sdk 1.4.6 was not available yet on maven central

**Comment by ddevadat** [comment]
@AmiableAnil @maheshkumargangula 
circle ci build keeps failing.  any pointers for this error

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:3.0.0-M5:test (default-test) on project lms-service: There are test failures.
[ERROR] 
[ERROR] Please refer to /home/circleci/project/service/target/surefire-reports for the individual test results.
[ERROR] Please refer to dump files (if any exist) [date].dump, [date]-jvmRun[N].dump and [date].dumpstream.
[ERROR] The forked VM terminated without properly saying goodbye. VM crash or System.exit called?
[ERROR] Command was /bin/sh -c cd /home/circleci/project/service && /usr/local/openjdk-14/bin/java -javaagent:/home/circleci/.m2/repository/org/jacoco/org.jacoco.agent/0.8.5/org.jacoco.agent-0.8.5-runtime.jar=destfile=/home/circleci/project/service/target/coverage-reports/jacoco-unit.exec -Xmx1024m -XX:MaxPermSize=256m org.apache.maven.surefire.booter.ForkedBooter /home/circleci/project/service/target/surefire 2023-05-03T11-31-35_352-jvmRun1 surefire2068639134003811541tmp surefire_05355295097836275673tmp
[ERROR] Error occurred in starting fork, check output in log
[ERROR] Process Exit Code: 137
[ERROR] Crashed tests:
[ERROR] util.RequestInterceptorTest
[ERROR] org.apache.maven.surefire.booter.SurefireBooterForkException: The forked VM terminated without properly saying goodbye. VM crash or System.exit called?
[ERROR] Command was /bin/sh -c cd /home/circleci/project/service && /usr/local/openjdk-14/bin/java -javaagent:/home/circleci/.m2/repository/org/jacoco/org.jacoco.agent/0.8.5/org.jacoco.agent-0.8.5-runtime.jar=destfile=/home/circleci/project/service/target/coverage-reports/jacoco-unit.exec -Xmx1024m -XX:MaxPermSize=256m org.apache.maven.surefire.booter.ForkedBooter /home/circleci/project/service/target/surefire 2023-05-03T11-31-35_352-jvmRun1 surefire2068639134003811541tmp surefire_05355295097836275673tmp
[ERROR] Error occurred in starting fork, check output in log
[ERROR] Process Exit Code: 137
[ERROR] Crashed tests:
[ERROR] util.RequestInterceptorTest
[ERROR] 	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.fork(ForkStarter.java:748)
[ERROR] 	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.run(ForkStarter.java:305)
[ERROR] 	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.run(ForkStarter.java:265)
[ERROR] 	at org.apache.maven.plugin.surefire.AbstractSurefireMojo.executeProvider(AbstractSurefireMojo.java:1314)
[ERROR] 	at org.apache.maven.plugin.surefire.AbstractSurefireMojo.executeAfterPreconditionsChecked(AbstractSurefireMojo.java:1159)
[ERROR] 	at org.apache.maven.plugin.surefire.AbstractSurefireMojo.execute(AbstractSurefireMojo.java:932)
[ERROR] 	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:137)
[ERROR] 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:210)
[ERROR] 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:156)
[ERROR] 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:148)
[ERROR] 	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:117)
[ERROR] 	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:81)
[ERROR] 	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:56)
[ERROR] 	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
[ERROR] 	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:305)
[ERROR] 	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:192)
[ERROR] 	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:105)
[ERROR] 	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:957)
[ERROR] 	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:289)
[ERROR] 	at org.apache.maven.cli.MavenCli.main(MavenCli.java:193)
[ERROR] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[ERROR] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[ERROR] 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[ERROR] 	at java.base/java.lang.reflect.Method.invoke(Method.java:564)
[ERROR] 	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:282)
[ERROR] 	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:225)
[ERROR] 	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:406)
[ERROR] 	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:347)
[ERROR] 
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <args> -rf :lms-service

Exited with code exit status 1
```



Is there any prerequsite to be met before running test locally.

File: course-mw/sunbird-util/sunbird-platform-core/common-util/src/main/java/org/sunbird/common/models/util/cloud/CloudService.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
+package org.sunbird.common.models.util.cloud;
+
+import java.io.File;
+import java.util.List;
+
+/** Created by arvind on 24/8/17. */
+public interface CloudService {
```
Comment (AmiableAnil) [inline course-mw/sunbird-util/sunbird-platform-core/common-util/src/main/java/org/sunbird/common/models/util/cloud/CloudService.java:7]: CloudService and CloudUtils, both the classes are not used any where. Can you delete it.

File: course-mw/sunbird-util/sunbird-platform-core/common-util/src/main/java/org/sunbird/common/util/CloudStorageUtil.java [hunk 4]
```diff
@@ -51,26 +51,34 @@ public static String getSignedUrl(
             Some.apply("r"), Some.apply("application/pdf"));
   }
 
-  private static BaseStorageService getStorageService(String storageType) {
-    String storageKey = PropertiesCache.getInstance().getProperty(JsonKey.ACCOUNT_NAME);
-    String storageSecret = PropertiesCache.getInstance().getProperty(JsonKey.ACCOUNT_KEY);
-    return getStorageService(storageType, storageKey, storageSecret);
-  }
+	public static String getAnalyticsSignedUrl(String storageType, String container, String objectKey) {
```
Comment (AmiableAnil) [inline course-mw/sunbird-util/sunbird-platform-core/common-util/src/main/java/org/sunbird/common/util/CloudStorageUtil.java:54]: Is this method required.

Comment (ddevadat) [reply course-mw/sunbird-util/sunbird-platform-core/common-util/src/main/java/org/sunbird/common/util/CloudStorageUtil.java:54]: this method (getAnalyticsSignedUrl) was there in the release tag release-5.0.0_RC2. I dont see this in the release-5.0.2 branch. Is it obsolete now

**Comment by subhashchandrab** [comment]
@AmiableAnil , Can this PR be merged now as all the review comments are fixed

File: course-mw/sunbird-util/sunbird-platform-core/common-util/src/main/java/org/sunbird/common/util/CloudStorageUtil.java (hunk 4 above)
Comment (ddevadat) [reply course-mw/sunbird-util/sunbird-platform-core/common-util/src/main/java/org/sunbird/common/util/CloudStorageUtil.java:54]: removed obsolete method getAnalyticsSignedUrl

**Review by AmiableAnil — APPROVED** [review]

**Pull request merged**